aws_lambda_powertools/utilities/idempotency/idempotency.py

```diff
@@ -43,20 +45,25 @@ def idempotent(
         Lambda's Context
     persistence_store: BasePersistenceLayer
         Instance of BasePersistenceLayer to store data
+    config: IdempotencyConfig
+        Configutation
 
     Examples
     --------
     **Processes Lambda's event in an idempotent manner**
-        >>> from aws_lambda_powertools.utilities.idempotency import idempotent, DynamoDBPersistenceLayer
+        >>> from aws_lambda_powertools.utilities.idempotency import (
+        >>>    idempotent, DynamoDBPersistenceLayer, IdempotencyConfig
+        >>> )
         >>>
-        >>> persistence_store = DynamoDBPersistenceLayer(event_key_jmespath="body", table_name="idempotency_store")
+        >>> idem_config=IdempotencyConfig(event_key_jmespath="body")
+        >>> persistence_layer = DynamoDBPersistenceLayer(table_name="idempotency_store")
         >>>
-        >>> @idempotent(persistence_store=persistence_store)
+        >>> @idempotent(config=idem_config, persistence_store=persistence_layer)
         >>> def handler(event, context):
         >>>     return {"StatusCode": 200}
     """
 
-    idempotency_handler = IdempotencyHandler(handler, event, context, persistence_store)
+    idempotency_handler = IdempotencyHandler(handler, event, context, config or IdempotencyConfig(), persistence_store)
```

Comment:
to make refactoring easier later, could you please (I can do too): Use `or` logic outside the instantiation e.g. config = config or IdempotencyConfig()

nitpick: kwargs over args for refactoring too